imported/w3c/web-platform-tests/IndexedDB/file_support.sub.html is flakily failing on wpt

szewai opened the pull request:
#### 2c8d8c5d73696aa0ebbda46a79725ea8f06a5250
<pre>
imported/w3c/web-platform-tests/IndexedDB/file_support.sub.html is flakily failing on wpt
<a href="https://bugs.webkit.org/show_bug.cgi?id=278242">https://bugs.webkit.org/show_bug.cgi?id=278242</a>
<a href="https://rdar.apple.com/134071413">rdar://134071413</a>

Reviewed by Brady Eidson.

The test expects the metadata of File object (including lastModified) retrieved from IndexedDB database is the same as
the original File object that gets stored into database. In existing implementation, we store metadata together with
File data on disk, and when we crete File object from disk data, we populate it with the metadata on disk. The test
fails as lastModified value is not restored correctly, becasue we only store lastModifiedOverride but not actual
lastModified value. This patch fixes this by ensuring lastModified value is written during JSValue serialization.

This patch does not update the test expectation because the test relies on WebDriver&apos;s functions and it still fails in
WebKitTestRunner.

* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneSerializer::write):

Canonical link: <a href="https://commits.webkit.org/282376@main">https://commits.webkit.org/282376@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c33cc21c7ccb0edc2d71249568b90911b7e93a18

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62957 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42313 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15553 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66978 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13561 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65077 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50000 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13845 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50755 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9361 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66026 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39321 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54526 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31437 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36011 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11859 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12437 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57556 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12189 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68673 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6903 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11818 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58071 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6935 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54586 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58271 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13972 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5765 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38133 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39213 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40324 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/38955 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/bindings/js/SerializedScriptValue.cpp:
```diff
@@ -2378,7 +2378,14 @@ class CloneSerializer : public CloneBase {
         write(file.type());
         write(file.name());
-        write(static_cast<double>(file.lastModifiedOverride().value_or(-1)));
+        if (m_forStorage == SerializationForStorage::No)
```
Can't review because the explanation in the commit log is a bit weak. Maybe @beidson can though.

Still I'd recommend writing proper commit logs that explains what the bug was and how it is now addressed.

szewai replied:
Updated.

szewai committed:
imported/w3c/web-platform-tests/IndexedDB/file_support.sub.html is flakily failing on wpt
https://bugs.webkit.org/show_bug.cgi?id=278242
rdar://134071413

Reviewed by Brady Eidson.

The test expects the metadata of File object (including lastModified) retrieved from IndexedDB database is the same as
the original File object that gets stored into database. In existing implementation, we store metadata together with
File data on disk, and when we crete File object from disk data, we populate it with the metadata on disk. The test
fails as lastModified value is not restored correctly, becasue we only store lastModifiedOverride but not actual
lastModified value. This patch fixes this by ensuring lastModified value is written during JSValue serialization.

This patch does not update the test expectation because the test relies on WebDriver's functions and it still fails in
WebKitTestRunner.

* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneSerializer::write):

Canonical link: https://commits.webkit.org/282376@main
